location/utils.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""
+    utils
+    ~~~~~
+
+    Utility functions for motif analysis.
+"""
+
+import pandas as pd
+import numpy as np
+import math
+from copy import deepcopy
+from collections import Counter
+from geopy.distance import vincenty
+import geohash
+
+from location import motif
+
+
+def approx_home_location(data, sr_col='stay_region'):
+    """
+    Calculate the home location based on the assumption that the most
+    frequently visited location during 24:00 to 6:00 is the home location.
+
+    Parameters:
+    -----------
+    data: Dataframe
+        User location data
+
+    sr_col: str
+        Column name for stay region.
+        Default is 'stay_region'.
+
+    Returns:
+    --------
+    home: str
+        Home location in geohash value.
+    """
+    loc_data = data.copy()
+    loc_data = loc_data[pd.notnull(loc_data[sr_col])]
+    loc_data['hour'] = [x.hour for x in loc_data.index]
+    # get the location data in the target hours
+    trg_data = loc_data.loc[loc_data['hour'].isin([0, 1, 2, 3, 4, 5])]
+    # if no data during 0:00 from 5:00, use location data from 20:00 to 8:00
+    if len(trg_data) == 0:
```

Comment:
I think you are right. We should be just fine to return None in this case. My only concern was that if, though very unlikely, one user has no location data between 0:00 - 6:00, then we wouldn't be able to get and insert the home location when generating motifs. In that case, we have to discard days that do not have any data in their first 30 min time slot. Of course it won't be a problem if we do not insert home location at all.